src/main/java/io/pravega/connectors/hadoop/PravegaClientConfig.java

```diff
@@ -0,0 +1,167 @@
+    static final PravegaParameter SCOPE_PARAM = new PravegaParameter("scope", "pravega.scope", "PRAVEGA_SCOPE");
+
+    private static final long serialVersionUID = 1L;
+
+    private URI controllerURI;
+    private String defaultScope;
+    private Credentials credentials;
+    private boolean validateHostname = true;
+    private String trustStore;
+
+    PravegaClientConfig(Properties properties, Map<String, String> env, Map<String, String> params) {
+        Preconditions.checkNotNull(properties, "Properties cannot be null");
+        Preconditions.checkNotNull(env, "env cannot be null");
+        Preconditions.checkNotNull(params, "params cannot be null");
+        this.controllerURI = CONTROLLER_PARAM.resolve(params, properties, env).map(URI::create).orElse(null);
+        this.defaultScope = SCOPE_PARAM.resolve(params, properties, env).orElse(null);
+    }
+
+    /**
+     * Gets a configuration based on defaults obtained from the local environment.
+     */
+    public static PravegaClientConfig fromDefaults() {
+        return new PravegaClientConfig(System.getProperties(), System.getenv(), new HashMap<>());
+    }
+
+    /**
+     * Gets a configuration based on defaults obtained from the local environment plus the given program parameters.
+     *
+     * @param params the parameters to use.
+     */
+    public static PravegaClientConfig fromParams(Map<String, String> params) {
+        return new PravegaClientConfig(System.getProperties(), System.getenv(), params);
+    }
+
+    /**
+     * Gets the {@link ClientConfig} to use with the Pravega client.
+     */
+    public ClientConfig getClientConfig() {
+        ClientConfig.ClientConfigBuilder builder = ClientConfig.builder()
+                .validateHostName(validateHostname);
+        if (controllerURI != null) {
+            builder.controllerURI(controllerURI);
+        }
+        if (credentials != null) {
+            builder.credentials(credentials);
+        }
+        if (trustStore != null) {
+            builder.trustStore(trustStore);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Configures the Pravega controller RPC URI.
+     *
+     * @param controllerURI The URI.
+     */
+    public PravegaClientConfig withControllerURI(URI controllerURI) {
+        this.controllerURI = controllerURI;
+        return this;
+    }
+
+    /**
+     * Configures truststore value.
+     * @param trustStore truststore name.
+     * @return current instance of PravegaClientConfig.
+     */
+    public PravegaClientConfig withTrustStore(String trustStore) {
+        this.trustStore = trustStore;
+        return this;
+    }
+
+    /**
+     * Configures the default Pravega scope, to resolve unqualified stream names and to support reader groups.
+     *
+     * @param scope The scope to use (with lowest priority).
+     */
+    public PravegaClientConfig withDefaultScope(String scope) {
```

Comment:
What if this method is called twice? The returned value will be the old scope, but what was passed was a new scope. IMO, this should be more deterministic such as: 
* Either, if current scope is not null throw an error 
* or, modify the current scope to the specified scope and returned the specified scope

Reply:
removed the validation and it will accept any value passed to the method.